src/key_gen/tests.rs

```diff
@@ -109,50 +110,88 @@ fn all_nodes_being_responsive() -> Result<()> {
+                        for proposal in proposal_vec {
+                            proposals.push(proposal);
+                        }
                     }
                 }
-            }
-        });
-        // Continue the procedure with messaging.
-        messaging(
-            &mut rng,
-            &mut generators,
-            &mut proposals,
-            non_responsives.clone(),
-        );
-        assert!(proposals.is_empty());
-    }
+            });
+            // Continue the procedure with messaging.
+            messaging(
+                &mut rng,
+                &mut generators,
+                &mut proposals,
+                non_responsive.clone(),
+            );
+            assert!(proposals.is_empty());
+        }
 
-    generators
-        .iter_mut()
-        .enumerate()
-        .for_each(|(index, key_gen)| {
-            if !non_responsives.contains(&(index as u64)) {
-                assert!(key_gen.generate_keys().is_some());
-                non_responsives.iter().for_each(|idx| {
+        let responsive = all_nodes.difference(&non_responsive).cloned().collect_vec();
+
+        let pub_key_set: PublicKeySet = generators[responsive.as_slice()[0] as usize]
+            .generate_keys()
+            .expect("Failed to generate `PublicKeySet` for node #0")
+            .1
+            .public_key_set;
+
+        let msg = "Test message!";
+        let mut sig_shares: BTreeMap<usize, SignatureShare> = BTreeMap::new();
+
+        for (index, key_gen) in generators.iter_mut().enumerate() {
+            if !non_responsive.contains(&(index as u64)) {
+                let outcome = if let Some(outcome) = key_gen.generate_keys() {
+                    outcome.1
+                } else {
+                    return Err(format_err!(
+                        "Failed to generate `PublicKeySet` and `SecretKeyShare` for node #{}",
+                        index
+                    ));
+                };
+                let sk = outcome.secret_key_share;
+                let index = key_gen.our_index as usize;
+                let pks = outcome.public_key_set;
+                assert_eq!(pks, pub_key_set);
+                let sig = sk.sign(msg);
+                assert!(pks.public_key_share(index).verify(&sig, msg));
+                let _ = sig_shares.insert(index, sig);
+
+                non_responsive.iter().for_each(|idx| {
                     assert!(!key_gen.names().contains(&peer_ids[*idx as usize].name()))
                 });
             } else {
                 assert!(key_gen.generate_keys().is_none());
-            }
-        });
+            };
+        }
+
+        let sig = match pub_key_set.combine_signatures(sig_shares.iter()) {
+            Ok(sig) => sig,
+            Err(e) => return Err(format_err!("Unexpected Error {:?}: Not able to generate Signature with THRESHOLD + 1 sig_shares", e)),
+        };
+
+        assert!(pub_key_set.public_key().verify(&sig, msg));
+    }
```

Comment:
Maybe I'm missing something, but we are still not checking whether the keys generated by the different subsets are all the same, do we?

Reply:
After a discussion, this probably doesn't make sense. Disregard.